runtime/src/main/java/org/corfudb/protocols/wireprotocol/failuredetector/FileSystemStats.java

```diff
@@ -14,6 +15,19 @@ public class FileSystemStats {
 
     @NonNull
     private final PartitionAttributeStats partitionAttributeStats;
+    @NonNull
+    private final BatchProcessorStats batchProcessorStats;
+
+    public boolean hasError() {
+        boolean isReadOnly = partitionAttributeStats.isReadOnly();
+        boolean isBpError = batchProcessorStats.isError();
+
+        return isReadOnly || isBpError;
+    }
+
+    public boolean isOk() {
```

Comment:
I think its cleaner to just have a generic category of LogUnit IO errors instead of having component specific errors, that information can be logged, but the information exchanged by failure detectors should be of a general category of faults.